Octokit/Http/Connection.cs

```diff
@@ -136,6 +136,30 @@ public Connection(ProductHeaderValue productInformation, Uri baseAddress, ICrede
             _jsonPipeline = new JsonHttpPipeline();
         }
 
+        /// <summary>
+        /// Gets the latest API Info - this will be null if no API calls have been made
+        /// </summary>
+        /// <returns><seealso cref="ApiInfo"/> representing the information returned as part of an Api call</returns>
+        public ApiInfo LastApiInfo
+        { 
+            get
+            {
+                lock (LastApiInfoLocker)
+                {
+                    return _lastApiInfo;
+                }
+            }
+            private set
+            {
+                lock (LastApiInfoLocker)
+                {
+                    _lastApiInfo = value;
+                }
+            }
+        }
+        private ApiInfo _lastApiInfo;
+        private readonly object LastApiInfoLocker = new object();
+
```

Comment:
I've been taking the Benevolent Dictator role. :stuck_out_tongue:

I'd like to hear from @khellang if there are specific concerns with removing the locks. If not, let's remove them.